File: System/MeleeGameSystem.lua
local SystemBase = require("System.SystemBase")

---@class MeleeGameSystem : SystemBase
local cls = class("MeleeGameSystem", SystemBase)

function cls:ctor()
    MeleeStartingVisibility()
    MeleeStartingHeroLimit()
    MeleeGrantHeroItems()
    MeleeStartingResources()
    MeleeClearExcessUnits()
    MeleeStartingUnits()
    MeleeStartingAI()
    MeleeInitVictoryDefeat()
end

return cls

File: CSProject/Systems/MeleeGameSystem.cs
using LuaWrapper;
using SFLib.Interop;

namespace Systems;

[Lua(Class = "MeleeGameSystem")]
public class MeleeGameSystem : SystemBase
{
    public MeleeGameSystem()
    {
        MeleeStartingVisibility();
        MeleeStartingHeroLimit();
        MeleeGrantHeroItems();
        MeleeStartingResources();
        MeleeClearExcessUnits();
        MeleeStartingUnits();
        MeleeStartingAI();
        MeleeInitVictoryDefeat();
    }
}
